### DISCUSSION TITLE
[Question]: Any paths available for deploying on Google App Engine? I am tripping up around local db support it seems when deploying.

### DISCUSSION THREAD

**Discussion by jordseast1** [danny-avila/LibreChat#2589] [Q&A] [answered]
### What is your question?

as title says, just interested in a path or some docs on how i could deploy this on google cloud engine. I have it working perfectly locally, but running into odd error like this when deploying: 

ERROR: gcloud crashed (MultiError): One or more errors occurred:
OperationalError: unable to open database file

OperationalError: unable to open database file

the only thing i added to .env file was the rag_openai_key and embedding variables. 

### More Details

thank you, what a great front end for these technologies!! 

### What is the main subject of your question?

Documentation, Installation

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
How are you installing it exactly? Does App engine let you run node.js and install "locally"?

**Reply by R-Prady** [reply]
https://cloud.google.com/appengine/docs/flexible

**Comment by jordseast1** [comment, marked as answer]
In a better place with this now. A couple things if anyone is going this route:

- need to go with an online mongo db
- still working on how we would deploy the RAG API portion, but a path to deploying with online mongodb looks clear. 

I am effectively following the local mac setup docs and then building a docker image and deploying on app engine. 

Love the docker compose setup for local use it was so easy to get rolling. 

Thx for moving to discussion @danny-avila , was really impressed w the product and want to get it deployed internally so if anyone out there has went the App Engine route please chime in or reach out with your experience! Would love to collaborate on some docs for this.

**Reply by R-Prady** [reply]
i'm trying to deploy it on GCP as well. and i was considering App engine. not sure where to start. would love a "getting started" guide from you
